src/lib/scheduler/topology.cpp

```diff
@@ -21,7 +21,15 @@ Topology& Topology::current() {
   return instance;
 }
 
-Topology::Topology() { init_default_topology(); }
+Topology::Topology()
+#if HYRISE_NUMA_SUPPORT
+  : _numa_num_configured_nodes(numa_num_configured_nodes())
+#else
+  : _numa_num_configured_nodes(1)
+#endif
+{
+  init_default_topology();
```

Comment:
Hmm, the control flow seems a bit unclear to me. I think calling `Topology::current().init_numa_topology` is a bit confusing, because I do not want to do anything with the current topology.

Maybe we should do it more like the `CurrentScheduler`, so that we can call `Topology::use_numa_topology` and `Topology::get()`?

Open for discussion.

Reply:
I'm fine with either one. @mrzzzrm any thoughts, since you were in on the previous PR about this?

Reply:
An interface like 
```c++
Topology::current();
Topology::use_numa_topology();
Topology::use_non_numa_topology();
...
```
seems reasonable to me.